### DISCUSSION TITLE
ionice: does not appear to work in firejail

### DISCUSSION THREAD

**Discussion by emerajid** [netblue30/firejail#6350] [Q&A] [answered]
How does firejail behave with ionice? The capability is often blocked inside the sandbox thus not available. Is starting firejail with ionice have any effect at all? Say,
```
ionice -c 3 firejail bash
```
Does this work as expected? Thus far my observations are that it doesn't. If I am correct, I suggest this gets implemented, because there are certain disk heavy frameworks which I'd rather ionice lest they do some obscene things to my HDD.

My observations are purely subjective and may be (severely) biased by disk swapping. Is there a reliable way to test this at all?

**Comment by rusty-snake** [comment, marked as answer]
> Thus far my observations are that it doesn't.
> My observations are purely subjective and may be (severely) biased by disk swapping. Is there a reliable way to test this at all?

How do you test.

"by feeling" is always a worst test method. Because for me:

```console
$ ionice
none: prio 0
$ firejail --quiet ionice 
none: prio 0
$ ionice -c 3 firejail --quiet ionice
idle
```

> suggest this gets implemented

If it is a kernel restriction (for example for SUID binaries), it can not magically be lifted. In that case run `ionice` inside firejail like `firejail --profile=foobar ionice -c3 foobar`.

**Comment by emerajid** [comment]
Thank you, that answers my question. Somehow I did not notice issuing `ionice` without arguments yields current priority. Perhaps I was just swapping and AFAIK ionice priority has nothing to do with swap.
